test/unit/http/request_test.dart

```diff
@@ -1037,6 +1039,57 @@ _runAutoRetryTestSuiteFor(
         expect(request.autoRetry.numAttempts, equals(4));
       });
 
+      test('exponential retry back-off with jitter', () async {
+        MockTransports.http.expect('GET', requestUri,
+            respondWith: new MockResponse.internalServerError());
+        MockTransports.http.expect('GET', requestUri,
+            respondWith: new MockResponse.internalServerError());
+        MockTransports.http.expect('GET', requestUri,
+            respondWith: new MockResponse.internalServerError());
+        MockTransports.http.expect('GET', requestUri);
+
+        BaseRequest request = requestFactory();
+        request.autoRetry
+          ..enabled = true
+          ..maxRetries = 3
+          ..backOff = new RetryBackOff.exponential(
+              new Duration(milliseconds: 25),
+              withJitter: true);
+        request.get(uri: requestUri);
+
+        // 1st attempt = immediate
+        // 2nd attempt = +0 to 50s (25*2^1)
+        // 3rd attempt = +0 to 100s (25*2^2) + 2nd attempt
+        // 4th attempt = +0 to 200s (25*2^3) + 3rd attempt
+        await new Future.delayed(new Duration(milliseconds: 350));
```

Comment:
Could you add another expectation just to assert that they're not all happening immediately? Something like

```dart
await new Future.delayed(new Duration(milliseconds: 50));
expect(request.autoRetry.numAttempts, greaterThan(0));
expect(request.autoRetry.numAttempts, lessThan(4));
await new Future.delayed(new Duration(milliseconds: 300));
expect(request.autoRetry.numAttempts, equals(4));
```